### PR TITLE
Added create link shortcut

### PR DESCRIPTION
**Pull request by adriantran**
- added create link shortcut
- added selectAll to LeanRooster
- fixed FocusOutShell command bar toggle submenu scenario
- updated tools/dts.js to handle const string and function with dictionary type

### CODE REVIEW

File: packages/roosterjs-react-command-bar/lib/components/RoosterCommandBar.tsx [hunk 1]
```diff
@@ -14,33 +14,36 @@ import './RoosterCommandBar.scss.g';
...
+        this._initButtonKeys(props);
+
         this._async = new Async();
         this._updateFormatStateDebounced = this._async.debounce(() => this._updateFormatState(), 100);
     }
 
     public render(): JSX.Element {
         const { className } = this.props;
 
+        const items = this._createItems(); // with the newest changes on the editor, get create the latest items
```
Comment (pranavgk) [inline packages/roosterjs-react-command-bar/lib/components/RoosterCommandBar.tsx]: nit: Not clear what you mean by "get create the latest items".

File: packages/roosterjs-react-command-bar/lib/plugins/RoosterCommandBarPlugin.ts [hunk 2]
```diff
@@ -14,24 +17,56 @@ export default class RoosterCommandBarPlugin implements EditorPlugin, RoosterCom
...
+            return;
+        }
+
+        if (event.eventType === PluginEventType.KeyDown) {
+            this.handleShortcuts(event);
+        }
+    }
+
+    private handleShortcuts(event: PluginEvent) {
+        let keyboardEvent = (event as PluginDomEvent).rawEvent as KeyboardEvent;
```
Comment (pranavgk) [inline packages/roosterjs-react-command-bar/lib/plugins/RoosterCommandBarPlugin.ts]: nit: const. Also, can event be undefined or null at any point?

File: packages/roosterjs-react-editor/lib/components/FocusOutShell.tsx [hunk 3]
```diff
@@ -48,6 +48,12 @@ export default class FocusOutShell extends React.PureComponent<FocusOutShellProp
         // For Callout component, target is the event object from the document.body focus event
         const nextTarget = ev && (ev.target as HTMLElement);
 
+        // command bar can trigger dismiss event w/o an event object for submenu (when 
+        // button is clicked again to hide submenu)
+        if (!ev) {
+            return;
+        }
```
Comment (pranavgk) [inline packages/roosterjs-react-editor/lib/components/FocusOutShell.tsx]: Move this check before assigning `nextTarget`

File: packages/roosterjs-react-command-bar/lib/plugins/RoosterCommandBarPlugin.ts (hunk 2 above)
Comment (adriantran) [reply packages/roosterjs-react-command-bar/lib/plugins/RoosterCommandBarPlugin.ts]: looking at attachDomEvent and DefaultShortcut, no

File: packages/roosterjs-react-editor/lib/components/FocusOutShell.tsx [hunk 4]
```diff
@@ -45,6 +45,12 @@ export default class FocusOutShell extends React.PureComponent<FocusOutShellProp
...
 
     private _calloutOnDismiss = (ev: React.FocusEvent<HTMLElement>): void => {
+        // command bar can trigger dismiss event w/o an event object for submenu (when 
+        // button is clicked again to hide submenu)
+        if (!ev) {
+            return;
+        }
+
         // For Callout component, target is the event object from the document.body focus event
         const nextTarget = ev && (ev.target as HTMLElement);
```
Comment (pranavgk) [inline packages/roosterjs-react-editor/lib/components/FocusOutShell.tsx]: nit: remove `ev`

**Pull request merged**